### PR TITLE
MM-32453 - Playbook can be edited to remove all the members

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Remove remove when one member to remove

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-32453

### CODE REVIEW

File: server/incident/service_test.go [hunk 1]
```diff
@@ -167,6 +167,7 @@ func TestCreateIncident(t *testing.T) {
 		}
 
 		store.EXPECT().CreateIncident(gomock.Any()).Return(incdnt, nil)
+		store.EXPECT().CreateTimelineEvent(gomock.AssignableToTypeOf(&incident.TimelineEvent{}))
```
Comment (lieut-data) [inline server/incident/service_test.go:170]: I've pushed https://github.com/mattermost/mattermost-plugin-incident-collaboration/pull/462 in a bid to correct this, but 0/5 on who gets merged first.

File: webapp/src/components/backstage/share_playbook.tsx [hunk 2]
```diff
@@ -57,7 +57,10 @@ const SharePlaybook: FC<SharePlaybookProps> = (props: SharePlaybookProps) => {
                 {props.playbook.member_ids.map((userId) => (
                     <UserLine key={userId}>
                         <SharePlaybookProfile userId={userId}/>
-                        <RemoveLink onClick={() => props.onRemoveUser(userId)} >{'Remove'}</RemoveLink>
+                        {
```
Comment (lieut-data) [inline webapp/src/components/backstage/share_playbook.tsx:60]: Worth an e2e test? (<ducks>)

**Review by lieut-data — APPROVED** [review]

File: server/incident/service_test.go (hunk 1 above)
Comment (cpoile) [reply server/incident/service_test.go:170]: @lieut-data I've actually fixed that in two PRs now. :)

File: webapp/src/components/backstage/share_playbook.tsx (hunk 2 above)
Comment (cpoile) [reply webapp/src/components/backstage/share_playbook.tsx:63]: ...fine. :P

File: tests-e2e/cypress/integration/backstage/playbook_creation_spec.js [hunk 3]
```diff
@@ -70,9 +73,36 @@ describe('playbook creation button', () => {
         // # Click 'Incident Collaboration Playbook'
         cy.findByText('Incident Collaboration Playbook').should('be.visible').click().wait(TIMEOUTS.TINY);
 
-        //Verify a new 'Incident Response Playbook' creation page is opened
+        // * Verify a new 'Incident Response Playbook' creation page is opened
         verifyPlaybookCreationPageOpened(url, playbookName);
     });
+
+    it('shows remove beside members when > 2 members', () => {
```
Comment (agarciamontoro) [inline tests-e2e/cypress/integration/backstage/playbook_creation_spec.js:80]:
```suggestion
    it('shows remove beside members when > 1 member', () => {
```

File: tests-e2e/cypress/integration/backstage/playbook_creation_spec.js [hunk 4]
```diff
@@ -70,9 +73,36 @@ describe('playbook creation button', () => {
...
+                .type('{enter}');
+
+            cy.findAllByTestId('user-line').should('have.length', 2);
+            cy.findAllByTestId('user-line').eq(0).within(() => {
+                cy.get('a').contains('Remove').should('exist');
+            });
+            cy.findAllByTestId('user-line').eq(1).within(() => {
+                cy.get('a').contains('Remove').should('exist');
+            });
+        });
```
Comment (agarciamontoro) [inline tests-e2e/cypress/integration/backstage/playbook_creation_spec.js:104]: If we have the `open backstage` comment, we should have these ones :joy: 

```suggestion
        cy.findByTestId('playbook-sidebar').within(() => {
            // * Verify that there is no Remove link when there is one member
            cy.findAllByTestId('user-line').should('have.length', 1);
            cy.findAllByTestId('user-line').eq(0).within(() => {
                cy.get('a').should('not.exist');
            });
           
            // # Add a new user
            cy.get('.profile-autocomplete__input > input')
                .type('anne stone', {force: true, delay: 100}).wait(100)
                .type('{enter}');

            // * Verify that there is a Remove link when there is more than one member
            cy.findAllByTestId('user-line').should('have.length', 2);
            cy.findAllByTestId('user-line').eq(0).within(() => {
                cy.get('a').contains('Remove').should('exist');
            });
            cy.findAllByTestId('user-line').eq(1).within(() => {
                cy.get('a').contains('Remove').should('exist');
            });
        });
```

**Review by agarciamontoro — APPROVED** [review]
Cool, thanks! I left a couple of non-blocking comments on comments :eyes:

**Review by prapti — APPROVED** [review]
Looks good to me. Manual and E2E tests both passed.

**Pull request merged**